collector/aws/elasticsearch_test.go

```diff
@@ -0,0 +1,159 @@
+package aws_test
+
+import (
+	"errors"
+	"finala/collector/aws"
+	"finala/collector/config"
+	"finala/collector/testutils"
+	"testing"
+
+	awsClient "github.com/aws/aws-sdk-go/aws"
+	elasticsearch "github.com/aws/aws-sdk-go/service/elasticsearchservice"
+)
+
+var defaultElasticSearchMock = elasticsearch.DescribeElasticsearchDomainsOutput{
+	DomainStatusList: []*elasticsearch.ElasticsearchDomainStatus{
+		{
+			ARN:        awsClient.String("arn-test"),
+			DomainName: awsClient.String("testDomain"),
+			ElasticsearchClusterConfig: &elasticsearch.ElasticsearchClusterConfig{
+				InstanceType:  awsClient.String("Type"),
+				InstanceCount: awsClient.Int64(2),
+			},
+			EBSOptions: &elasticsearch.EBSOptions{
+				EBSEnabled: awsClient.Bool(true),
+				VolumeSize: awsClient.Int64(10),
+				VolumeType: awsClient.String("gp2"),
+			},
+		},
+	},
+}
+
+type MockAWSElasticSearchClient struct {
+	responseDescribeClusters *elasticsearch.DescribeElasticsearchDomainsOutput
+	err                      error
+}
+
+func (es *MockAWSElasticSearchClient) DescribeElasticsearchDomains(*elasticsearch.DescribeElasticsearchDomainsInput) (*elasticsearch.DescribeElasticsearchDomainsOutput, error) {
+	return es.responseDescribeClusters, es.err
+}
+
+func (es *MockAWSElasticSearchClient) ListDomainNames(*elasticsearch.ListDomainNamesInput) (*elasticsearch.ListDomainNamesOutput, error) {
+	return &elasticsearch.ListDomainNamesOutput{
+		DomainNames: []*elasticsearch.DomainInfo{
+			{
+				DomainName: awsClient.String("testDomain"),
+			},
+		},
+	}, es.err
+}
+
+func (es *MockAWSElasticSearchClient) ListTags(*elasticsearch.ListTagsInput) (*elasticsearch.ListTagsOutput, error) {
+
+	return &elasticsearch.ListTagsOutput{}, es.err
+}
+
+func TestDescribeElasticSearchClusters(t *testing.T) {
+
+	collector := testutils.NewMockCollector()
+	metrics := []config.MetricConfig{}
+
+	t.Run("valid", func(t *testing.T) {
+
+		mockClient := MockAWSElasticSearchClient{
+			responseDescribeClusters: &defaultElasticSearchMock,
+		}
+
+		esManager := aws.NewElasticSearchManager(collector, &mockClient, nil, nil, nil, metrics, "us-east-1")
+
+		result, _ := esManager.DescribeClusters()
+
+		if len(result) != len(defaultElasticSearchMock.DomainStatusList) {
+			t.Fatalf("unexpected elasticsearch clusters count, got %d expected %d", len(result), len(defaultElasticSearchMock.DomainStatusList))
+		}
+	})
+
+	t.Run("error", func(t *testing.T) {
+
+		mockClient := MockAWSElasticSearchClient{
+			responseDescribeClusters: &defaultElasticSearchMock,
+			err:                      errors.New("error"),
+		}
+
+		esManager := aws.NewElasticSearchManager(collector, &mockClient, nil, nil, nil, metrics, "us-east-1")
+
+		_, err := esManager.DescribeClusters()
```

Comment:
you can check the `len` of cluster response

Reply:
done.

Reply:
added a check